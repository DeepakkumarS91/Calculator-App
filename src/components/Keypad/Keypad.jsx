import React from 'react';
import PropTypes from 'prop-types';
import Key from '../Key/Key';
import './Keypad.css';

const Keypad = ({numbers,operators,setOperator,callOperator,updateDisplay}) => {

    const numberKeys=numbers.map(number=><Key key={number} 
    keyValue={number}
    keyType='number-key'
    keyAction={updateDisplay}
    />);
    const operatorKeys=operators.map(operator=><Key key={operator}
    keyValue={operator}
    keyType='operator-key'
    keyAction={setOperator}
    />)
    return ( 
        <div className='keypad-container'>
            <div className="numbers-container">
                {numberKeys}
                </div>
                <div className='operators-container'>
                    {operatorKeys}
                    </div>
                    <div className="submit-container">
                    <Key
                    keyAction={callOperator}
                    keyValue='='
                    keyType='submit-key'
                    /></div>
        </div>
     );
}

Keypad.propTypes={
numbers:PropTypes.array.isRequired,
operators:PropTypes.array.isRequired,
setOperator:PropTypes.func.isRequired,
callOperator:PropTypes.func.isRequired,
updateDisplay:PropTypes.func.isRequired
}

export default Keypad;