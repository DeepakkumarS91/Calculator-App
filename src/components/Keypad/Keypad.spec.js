import React from 'react';
import Keypad from './Keypad';
import {shallow,mount} from 'enzyme';


describe('checking the Keypad component',()=>{
    let wrapper;
    beforeEach(()=>wrapper=shallow(<Keypad numbers={[]}
    operators={[]}
    callOperator={jest.fn()}
    setOperator={jest.fn()}
    updateDisplay={jest.fn()}/>));

    it('Snapshot of the Keypad component',()=>
    expect(wrapper).toMatchSnapshot()
    );

    it('checks for div',()=>
    expect(wrapper.find('div').length).toEqual(4));

    
        it('should render Key compoenent for each instance of numbers,operators and submit',()=>{
            let numbers=['1','2'];
            let operators=['+','-'];
            let submit=1;
            let totalKeys=numbers.length+operators.length+submit;
            wrapper.setProps({numbers,operators});
            expect(wrapper.find('Key').length).toEqual(totalKeys);
        })

    });

   describe('Mounted Keypad', ()=>{
     let wrapper;
     beforeEach(()=>{
        wrapper=mount(
        <Keypad
        numbers={[]}
        operators={[]}
        callOperator={jest.fn()}
        setOperator={jest.fn()}
        updateDisplay={jest.fn()}/>
        ); })

        it('Should render correctly',()=>
        expect(wrapper).toMatchSnapshot()
        );
    

    it('checks for the numbers props value',()=>
    {
        wrapper.setProps({numbers:['1','2','3']});
        expect(wrapper.find('.numbers-container').text()).toEqual('123');
    });

    it('checks for operators',()=>{
        wrapper.setProps({operators:['+','-','*','/']});
        expect(wrapper.find('.operators-container').text()).toEqual('+-*/');
    })    
