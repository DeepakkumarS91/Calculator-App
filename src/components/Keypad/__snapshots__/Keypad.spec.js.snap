// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted Keypad Should render correctly 1`] = `ReactWrapper {}`;

exports[`checking the Keypad component Snapshot of the Keypad component 1`] = `ShallowWrapper {}`;
